pkg/util/log/middleware.go

```diff
@@ -2,22 +2,41 @@ package log
 
 import (
 	"bufio"
+	"context"
 	"fmt"
 	"net"
 	"net/http"
 	"strings"
 	"time"
 
 	kitlog "github.com/go-kit/kit/log"
+	uuid "github.com/satori/go.uuid"
 )
 
+const (
+	KubernikusRequestID = "Kubernikus-Request-Id"
+)
+
+func RequestIDHandler(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(rw http.ResponseWriter, request *http.Request) {
+		if id := request.Context().Value(KubernikusRequestID); id == nil {
+			request = request.WithContext(context.WithValue(request.Context(), KubernikusRequestID, uuid.NewV4()))
```

Comment:
For my taste there we should have the world `context` in this line more often. ;)

Reply:
i resisted the temptation to make this wish come true ;)